### PR TITLE
SSCS-10853 Postponements Letter Templates Implementation

### PR DESCRIPTION
**Pull request by TeaWithLucas**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-10853

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by ss-evoco — APPROVED** [review]

File: build.gradle [hunk 1]
```diff
@@ -229,8 +229,8 @@ dependencyManagement {
             entry 'spring-cloud-openfeign-core'
         }
 
-        //CVE-2022-25857
-        dependencySet(group: 'org.yaml', version: '1.31') {
+        //CVE-2022-25857, CVE-2022-38751
+        dependencySet(group: 'org.yaml', version: '1.32') {
```
Comment (filippo-possenti-version1) [inline build.gradle:233]: The corresponding update has been done a few hours ago. A rebase should make this change obsolete.

File: src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/config/BundledNotificationsConfigTestIt.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        ISSUE_ADJOURNMENT_NOTICE_WELSH
+    );
+    private static final Set<AppealHearingType> APPEAL_HEARING_TYPES = Set.of(PAPER, ORAL);
+
+    @Test
+    @Parameters(method = "bundledLetterTemplateNames")
+    public void given_bundledLetters_should_notHaveTemplate(NotificationEventType eventType) {
+        List<Template> templates = getTemplates(eventType);
+
+        assertThat(templates).allSatisfy(template -> {
```
Comment (filippo-possenti-version1) [inline src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/config/BundledNotificationsConfigTestIt.java:45]: This test will succeed if no templates are returned.

Is this intended or should there be a check that there is at least one template?

File: src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/config/BundledNotificationsConfigTestIt.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        for (SubscriptionType subscriptionType : SubscriptionType.values()) {
+            for (AppealHearingType appealHearingType : APPEAL_HEARING_TYPES) {
+                templates.add(getTemplate(eventType, subscriptionType, appealHearingType, true, null));
+                templates.add(getTemplate(eventType, subscriptionType, appealHearingType, false, null));
+            }
+        }
+        return templates;
+    }
+
+    @SuppressWarnings({"unused"})
```
Comment (filippo-possenti-version1) [inline src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/config/BundledNotificationsConfigTestIt.java:65]: Why is this `@SuppressWarning` necessary?

File: src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/config/DirectionNoticeDocmosisConfigTestIt.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        assertNull(templateAppellant.getEmailTemplateId());
+        assertTrue(templateAppellant.getSmsTemplateId().isEmpty());
+        assertNull(templateAppellant.getLetterTemplateId());
+        assertEquals(configAppellantOrAppointee, templateAppellant.getDocmosisTemplateId());
+        Template templateAppointee = personalisation.getTemplate(wrapper, PIP, APPOINTEE);
+        assertEquals(configAppellantOrAppointee, templateAppointee.getDocmosisTemplateId());
+        Template templateRep = personalisation.getTemplate(wrapper, PIP, REPRESENTATIVE);
+        assertEquals(configRep, templateRep.getDocmosisTemplateId());
+        Template templateJointParty = personalisation.getTemplate(wrapper, PIP, JOINT_PARTY);
+        assertEquals(configJointParty, templateJointParty.getDocmosisTemplateId());
```
Comment (filippo-possenti-version1) [inline src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/config/DirectionNoticeDocmosisConfigTestIt.java:66]: You make such a good use of `assertThat` all around. I couldn't help but notice that you didn't use it here. Any specific reason?

File: src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/config/BundledNotificationsConfigTestIt.java (hunk 2 above)
Comment (TeaWithLucas) [reply src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/config/BundledNotificationsConfigTestIt.java:45]: Sure, I can add this

File: src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/config/BundledNotificationsConfigTestIt.java (hunk 3 above)
Comment (TeaWithLucas) [reply src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/config/BundledNotificationsConfigTestIt.java:65]: Its an annoyance with junit 4's parameter runner, I would really like to see all repos upgraded to jypter 5 tbh.
Without it causes a sonarLint code smell:
image

File: src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/config/DirectionNoticeDocmosisConfigTestIt.java (hunk 4 above)
Comment (TeaWithLucas) [reply src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/config/DirectionNoticeDocmosisConfigTestIt.java:66]: I was just moving methods, I'll do this, its not a big change

File: src/e2e/java/uk/gov/hmcts/reform/sscs/functional/sya/notifications/DocmosisWithGovNotifyLetterFunctionalTest.java [hunk 5]
```diff
@@ -1,40 +1,91 @@
...
         super(30);
     }
 
     @Test
     @Parameters(method = "eventTypes")
-    public void shouldSendDocmosisLetters(NotificationEventType notificationEventType, int expectedNumberOfLetters) throws IOException, NotificationClientException {
+    public void shouldSendDocmosisLetters(NotificationEventType notificationEventType) {
+        simulateCcdCallback(notificationEventType);
+
+        assertThatNoException().isThrownBy(() -> {
```
Comment (filippo-possenti-version1) [inline src/e2e/java/uk/gov/hmcts/reform/sscs/functional/sya/notifications/DocmosisWithGovNotifyLetterFunctionalTest.java:51]: Minor:

I liked it better with the `throws` declaration in this case.

The `assertThatNoException` is very useful, but not in this case. It's handy when you want to have multiple checks within the same test, especially if you want to check something AFTER the exception has been thrown (like some recording has been written for auditing purposes or things like that).

In this specific case it has no benefit: we really want to check that the letter subjects are correct. An exception being thrown is merely some other kind of problem that we didn't expect, not an explicit assertion to be checked.

File: src/e2e/java/uk/gov/hmcts/reform/sscs/functional/sya/notifications/DocmosisWithGovNotifyLetterFunctionalTest.java [hunk 6]
```diff
@@ -1,40 +1,91 @@
...
+    @Test
+    @Parameters(method = "expectedNumberOfLetters")
+    public void shouldSendCorrectNumberOfDocmosisLetters(NotificationEventType notificationEventType,
+                                                         int expectedNumberOfLetters) {
+        simulateCcdCallback(notificationEventType);
 
-        List<Notification> notifications = fetchLetters();
-        assertEquals(expectedNumberOfLetters, notifications.size());
-        notifications.forEach(n -> assertEquals("Pre-compiled PDF", n.getSubject().orElse("Unknown Subject")));
+        assertThatNoException().isThrownBy(() -> {
```
Comment (filippo-possenti-version1) [inline src/e2e/java/uk/gov/hmcts/reform/sscs/functional/sya/notifications/DocmosisWithGovNotifyLetterFunctionalTest.java:69]: Same as above on personal taste on the use of `assertThatNoException`.

File: src/main/java/uk/gov/hmcts/reform/sscs/factory/PersonalisationFactory.java [hunk 7]
```diff
@@ -28,54 +31,18 @@ public class PersonalisationFactory implements Function<NotificationEventType, P
...
-            ) {
-                selectedPersonalisation = withRepresentativePersonalisation;
-            } else if (SUBSCRIPTION_UPDATED_NOTIFICATION.equals(notificationType)) {
-                selectedPersonalisation = subscriptionPersonalisation;
-            } else {
-                selectedPersonalisation = this.personalisation;
-            }
+        if (isNull(notificationType)) {
+            return null;
+        }
```
Comment (filippo-possenti-version1) [inline src/main/java/uk/gov/hmcts/reform/sscs/factory/PersonalisationFactory.java:36]: Yes! Very much yes!
Short-circuiting helps a lot improving readability. Well done here!

File: src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/config/DirectionNoticeDocmosisConfigTestIt.java [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
```
[the remark this answers is not in the document]
Comment (TeaWithLucas) [reply src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/config/DirectionNoticeDocmosisConfigTestIt.java:105]: As above

File: src/main/java/uk/gov/hmcts/reform/sscs/service/NotificationService.java [hunk 9]
```diff
@@ -309,39 +319,24 @@ private String getSubscriptionDetails(String newSubscription, String oldSubscrip
...
                                                          NotificationEventType notificationType) {
-        if (REQUEST_INFO_INCOMPLETE.equals(notificationType)) {
-            if (StringUtils.isEmpty(notificationWrapper.getNewSscsCaseData().getInformationFromAppellant())
-                    || "No".equalsIgnoreCase(notificationWrapper.getNewSscsCaseData().getInformationFromAppellant())) {
-
-                log.info("Request Incomplete Information with empty or no Information From Appellant for ccdCaseId {}.", notificationWrapper.getNewSscsCaseData().getCcdCaseId());
-                return false;
-            }
+        if (REQUEST_INFO_INCOMPLETE.equals(notificationType)
+            && isNoOrNull(notificationWrapper.getNewSscsCaseData().getInformationFromAppellant())) {
```
Comment (filippo-possenti-version1) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/NotificationService.java:323]: The behaviour here is slightly different than before. You might want to double-check it.

Besides the check for "No", the StringUtils.isEmpty checks both for `null` and an empty string `""`. This means that you are effectively treating an empty string as "not no".

What is supposed to happen if `getInformationFromAppellant` returns an empty string? Is it even possible?

File: src/test/java/uk/gov/hmcts/reform/sscs/service/OutOfHoursCalculatorTest.java [hunk 10]
```diff
@@ -55,7 +55,7 @@ public void getStartOfNextInHoursPeriodWhenItIsTheNextDay() throws NoSuchAlgorit
         assertThat(nextInHoursTime.getMonthValue(), is(9));
         assertThat(nextInHoursTime.getDayOfMonth(), is(19));
         assertThat(nextInHoursTime.getHour(), is(START_HOUR));
-        assertThat(new Double(nextInHoursTime.getMinute()), closeTo(0, 59));
+        assertThat(Double.valueOf(nextInHoursTime.getMinute()), closeTo(0, 59));
```
Comment (filippo-possenti-version1) [inline src/test/java/uk/gov/hmcts/reform/sscs/service/OutOfHoursCalculatorTest.java:58]: I appreciate the effort... however it was bad before and it still is. I would suggest either to leave it alone or fixing its core problem.

`closeTo` in this case effectively means that any value between -59 and 59 will be accepted. Of course that's not what's intended here. What was intended was to validate that the value is between 0 and 59.

If you really want to fix it, try this:
`assertThat(nextInHoursTime.getMinute(), in(IntStream.range(0, 59).boxed().collect(Collectors.toSet())));`

File: src/test/java/uk/gov/hmcts/reform/sscs/service/OutOfHoursCalculatorTest.java [hunk 11]
```diff
@@ -67,7 +67,7 @@ public void getStartOfNextInHoursPeriodWhenItIsTheSameDay() throws NoSuchAlgorit
         assertThat(nextInHoursTime.getMonthValue(), is(9));
         assertThat(nextInHoursTime.getDayOfMonth(), is(18));
         assertThat(nextInHoursTime.getHour(), is(START_HOUR));
-        assertThat(new Double(nextInHoursTime.getMinute()), closeTo(0, 59));
+        assertThat(Double.valueOf(nextInHoursTime.getMinute()), closeTo(0, 59));
```
Comment (filippo-possenti-version1) [inline src/test/java/uk/gov/hmcts/reform/sscs/service/OutOfHoursCalculatorTest.java:70]: Same as above.

**Review by filippo-possenti-version1 — COMMENTED** [review]
No objections here, but neither full approval.

See if you can address one of the potential issues I found on a slight change in the way a condition works and provide an answer: that's the only one that might drive an approval or rejection but I don't have insights on the acceptable input values. Except for that, I don't have any specific objection, only general advice.

File: src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/config/BundledNotificationsConfigTestIt.java (hunk 3 above)
Comment (filippo-possenti-version1) [reply src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/config/BundledNotificationsConfigTestIt.java]: I see. Makes sense.

File: src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/config/DirectionNoticeDocmosisConfigTestIt.java (hunk 4 above)
Comment (filippo-possenti-version1) [reply src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/config/DirectionNoticeDocmosisConfigTestIt.java]: Ah, ok. I must have missed that you were not adding a test but just moving it.

No need to fix it then. I'll leave it up to you. It's already a big PR, no reason to make it bigger.

File: src/main/java/uk/gov/hmcts/reform/sscs/config/NotificationEventTypeLists.java [hunk 12]
```diff
@@ -0,0 +1,383 @@
+package uk.gov.hmcts.reform.sscs.config;
+
+import static uk.gov.hmcts.reform.sscs.domain.notify.NotificationEventType.*;
```
Comment (amoslv1) [inline src/main/java/uk/gov/hmcts/reform/sscs/config/NotificationEventTypeLists.java:3]: Surprised this * import has passed checkstyle test

File: src/main/java/uk/gov/hmcts/reform/sscs/config/NotificationEventTypeLists.java [hunk 13]
```diff
@@ -0,0 +1,383 @@
...
+        HEARING_REMINDER_NOTIFICATION,
+        ISSUE_ADJOURNMENT_NOTICE,
+        ISSUE_ADJOURNMENT_NOTICE_WELSH,
+        POSTPONEMENT_NOTIFICATION,
+        PROCESS_AUDIO_VIDEO,
+        PROCESS_AUDIO_VIDEO_WELSH,
+        STRUCK_OUT
+    );
+
+    public static final Set<NotificationEventType> EVENTS_VALID_FOR_FOR_APPOINTEE = EnumSet.of(
```
Comment (amoslv1) [inline src/main/java/uk/gov/hmcts/reform/sscs/config/NotificationEventTypeLists.java:215]: Is double "FOR" intentional? Here and the next one

File: src/main/java/uk/gov/hmcts/reform/sscs/config/NotificationEventTypeLists.java [hunk 14]
```diff
@@ -0,0 +1,383 @@
...
+        NON_COMPLIANT_NOTIFICATION,
+        PROVIDE_APPOINTEE_DETAILS,
+        RESEND_APPEAL_CREATED_NOTIFICATION,
+        SYA_APPEAL_CREATED_NOTIFICATION,
+        SUBSCRIPTION_UPDATED_NOTIFICATION,
+        TCW_DECISION_APPEAL_TO_PROCEED,
+        VALID_APPEAL_CREATED
+    );
+
+    public static final Set<NotificationEventType> EVENTS_VALID_FOR_FOR_REP = EnumSet.of(
```
Comment (amoslv1) [inline src/main/java/uk/gov/hmcts/reform/sscs/config/NotificationEventTypeLists.java:238]: Double "FOR"

File: src/main/java/uk/gov/hmcts/reform/sscs/personalisation/SyaAppealCreatedAndReceivedPersonalisation.java [hunk 15]
```diff
@@ -1,7 +1,7 @@
 package uk.gov.hmcts.reform.sscs.personalisation;
 
-import static uk.gov.hmcts.reform.sscs.config.AppConstants.*;
 import static uk.gov.hmcts.reform.sscs.config.PersonalisationConfiguration.PersonalisationKey.*;
+import static uk.gov.hmcts.reform.sscs.config.PersonalisationMappingConstants.*;
```
Comment (amoslv1) [inline src/main/java/uk/gov/hmcts/reform/sscs/personalisation/SyaAppealCreatedAndReceivedPersonalisation.java:4]: * Import

File: src/main/java/uk/gov/hmcts/reform/sscs/service/NotificationService.java [hunk 16]
```diff
@@ -309,39 +319,24 @@ private String getSubscriptionDetails(String newSubscription, String oldSubscrip
...
-                    || REISSUE_DOCUMENT.equals(notificationType)
-                    || PROVIDE_APPOINTEE_DETAILS.equals(notificationType))) {
-                log.info(format("Cannot complete notification %s as the appeal was dormant for caseId %s.",
-                        notificationType.getId(), notificationWrapper.getCaseId()));
-                return false;
-            }
+            && notificationWrapper.getSscsCaseDataWrapper().getState().equals(State.DORMANT_APPEAL_STATE)
+            && !EVENT_TYPES_NOT_FOR_DORMANT_CASES.contains(notificationType)) {
+            log.info(format("Cannot complete notification %s as the appeal was dormant for caseId %s.",
+                notificationType.getId(), notificationWrapper.getCaseId()));
```
Comment (amoslv1) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/NotificationService.java:333]:
```suggestion
            log.info("Cannot complete notification {} as the appeal was dormant for caseId {}.",
                notificationType.getId(), notificationWrapper.getCaseId());
```
Should be able to remove format

File: src/main/java/uk/gov/hmcts/reform/sscs/service/NotificationService.java [hunk 17]
```diff
@@ -309,39 +319,24 @@ private String getSubscriptionDetails(String newSubscription, String oldSubscrip
...
+            return false;
         }
-        if (notificationWrapper.getNewSscsCaseData().isLanguagePreferenceWelsh() && (ISSUE_FINAL_DECISION.equals(notificationType) || DECISION_ISSUED.equals(notificationType) || DIRECTION_ISSUED.equals(notificationType) || ISSUE_ADJOURNMENT_NOTICE.equals(notificationType) || PROCESS_AUDIO_VIDEO.equals(notificationType) || ACTION_POSTPONEMENT_REQUEST.equals(notificationType))) {
-            log.info(format("Cannot complete notification %s as the appeal is Welsh  for caseId %s.",
+
+        if (notificationWrapper.getNewSscsCaseData().isLanguagePreferenceWelsh()
+            && (EVENT_TYPES_NOT_FOR_WELSH_CASES.contains(notificationType))) {
+            log.info(format("Cannot complete notification %s as the appeal is Welsh for caseId %s.",
```
Comment (amoslv1) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/NotificationService.java:339]: Same here
```suggestion
            log.info("Cannot complete notification {} as the appeal is Welsh for caseId {}.",
```

File: src/main/java/uk/gov/hmcts/reform/sscs/service/NotificationService.java [hunk 18]
```diff
@@ -355,27 +350,38 @@ private boolean isEventAllowedToProceedWithValidData(NotificationWrapper notific
...
 
-        if (covid19Feature && (HEARING_BOOKED_NOTIFICATION.equals(notificationType) || HEARING_REMINDER_NOTIFICATION.equals(notificationType))) {
-            log.info(format("Notification not valid to send as covid 19 feature flag on for case id %s and event %s in state %s", notificationWrapper.getCaseId(), notificationType.getId(), notificationWrapper.getSscsCaseDataWrapper().getState()));
+        if (covid19Feature
+            && (HEARING_BOOKED_NOTIFICATION.equals(notificationType)
+            || HEARING_REMINDER_NOTIFICATION.equals(notificationType))) {
+            log.info(format("Notification not valid to send as covid 19 feature flag on for case id %s and event %s in state %s",
+                notificationWrapper.getCaseId(),
+                notificationType.getId(),
+                notificationWrapper.getSscsCaseDataWrapper().getState()));
```
Comment (amoslv1) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/NotificationService.java:373]:
```suggestion
            log.info("Notification not valid to send as covid 19 feature flag on for case id {} and event {} in state {}",
                notificationWrapper.getCaseId(),
                notificationType.getId(),
                notificationWrapper.getSscsCaseDataWrapper().getState());
```

File: src/main/java/uk/gov/hmcts/reform/sscs/service/NotificationService.java [hunk 19]
```diff
@@ -355,27 +350,38 @@ private boolean isEventAllowedToProceedWithValidData(NotificationWrapper notific
...
             return false;
         }
-        log.info(format("Notification valid to send for case id %s and event %s in state %s", notificationWrapper.getCaseId(), notificationType.getId(), notificationWrapper.getSscsCaseDataWrapper().getState()));
+        log.info(format("Notification valid to send for case id %s and event %s in state %s",
+            notificationWrapper.getCaseId(),
+            notificationType.getId(),
+            notificationWrapper.getSscsCaseDataWrapper().getState()));
```
Comment (amoslv1) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/NotificationService.java:379]:
```suggestion
        log.info(format("Notification valid to send for case id {} and event {} in state {}",
            notificationWrapper.getCaseId(),
            notificationType.getId(),
            notificationWrapper.getSscsCaseDataWrapper().getState());
```

File: src/main/java/uk/gov/hmcts/reform/sscs/service/SendNotificationService.java [hunk 20]
```diff
@@ -1,25 +1,44 @@
...
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+import static uk.gov.hmcts.reform.sscs.ccd.callback.DocumentType.ADJOURNMENT_NOTICE;
+import static uk.gov.hmcts.reform.sscs.ccd.callback.DocumentType.AUDIO_VIDEO_EVIDENCE_DIRECTION_NOTICE;
+import static uk.gov.hmcts.reform.sscs.ccd.callback.DocumentType.DECISION_NOTICE;
+import static uk.gov.hmcts.reform.sscs.ccd.callback.DocumentType.DIRECTION_NOTICE;
+import static uk.gov.hmcts.reform.sscs.ccd.callback.DocumentType.FINAL_DECISION_NOTICE;
 import static uk.gov.hmcts.reform.sscs.ccd.callback.DocumentType.POSTPONEMENT_REQUEST_DIRECTION_NOTICE;
-import static uk.gov.hmcts.reform.sscs.config.AppConstants.*;
+import static uk.gov.hmcts.reform.sscs.config.PersonalisationMappingConstants.*;
 import static uk.gov.hmcts.reform.sscs.domain.notify.NotificationEventType.*;
```
Comment (amoslv1) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/SendNotificationService.java:12]: These are * imports

File: src/test/java/uk/gov/hmcts/reform/sscs/personalisation/PersonalisationTest.java [hunk 21]
```diff
@@ -26,6 +26,8 @@
 import static uk.gov.hmcts.reform.sscs.ccd.domain.HearingType.REGULAR;
 import static uk.gov.hmcts.reform.sscs.ccd.domain.YesNo.YES;
 import static uk.gov.hmcts.reform.sscs.config.AppConstants.*;
+import static uk.gov.hmcts.reform.sscs.config.PersonalisationConfiguration.*;
+import static uk.gov.hmcts.reform.sscs.config.PersonalisationMappingConstants.*;
```
Comment (amoslv1) [inline src/test/java/uk/gov/hmcts/reform/sscs/personalisation/PersonalisationTest.java:30]: . * imports

File: src/test/java/uk/gov/hmcts/reform/sscs/personalisation/SubscriptionPersonalisationTest.java [hunk 22]
```diff
@@ -7,6 +7,7 @@
 import static org.mockito.Mockito.when;
 import static org.mockito.MockitoAnnotations.openMocks;
 import static uk.gov.hmcts.reform.sscs.ccd.domain.Benefit.PIP;
+import static uk.gov.hmcts.reform.sscs.config.PersonalisationMappingConstants.*;
 import static uk.gov.hmcts.reform.sscs.config.SubscriptionType.APPELLANT;
 import static uk.gov.hmcts.reform.sscs.domain.notify.NotificationEventType.*;
```
Comment (amoslv1) [inline src/test/java/uk/gov/hmcts/reform/sscs/personalisation/SubscriptionPersonalisationTest.java:12]: .* imports

File: src/test/java/uk/gov/hmcts/reform/sscs/personalisation/WithRepresentativePersonalisationTest.java [hunk 23]
```diff
@@ -1,6 +1,7 @@
 package uk.gov.hmcts.reform.sscs.personalisation;
 
 import static org.junit.Assert.assertEquals;
+import static uk.gov.hmcts.reform.sscs.config.PersonalisationMappingConstants.*;
```
Comment (amoslv1) [inline src/test/java/uk/gov/hmcts/reform/sscs/personalisation/WithRepresentativePersonalisationTest.java:4]: .*

File: src/test/java/uk/gov/hmcts/reform/sscs/personalisation/SyaAppealCreatedAndReceivedPersonalisationTest.java [hunk 24]
```diff
@@ -4,6 +4,26 @@
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNull;
 import static org.mockito.MockitoAnnotations.openMocks;
+import static uk.gov.hmcts.reform.sscs.config.PersonalisationConfiguration.PersonalisationKey.*;
```
Comment (amoslv1) [inline src/test/java/uk/gov/hmcts/reform/sscs/personalisation/SyaAppealCreatedAndReceivedPersonalisationTest.java:7]: .*

File: src/main/java/uk/gov/hmcts/reform/sscs/personalisation/Personalisation.java [hunk 25]
```diff
@@ -2,54 +2,92 @@
...
+import static uk.gov.hmcts.reform.sscs.config.AppConstants.THE_STRING_WELSH;
+import static uk.gov.hmcts.reform.sscs.config.AppConstants.TOMORROW_STRING;
+import static uk.gov.hmcts.reform.sscs.config.NotificationEventTypeLists.EVENTS_WITH_SUBSCRIPTION_TYPE_DOCMOSIS_TEMPLATES;
+import static uk.gov.hmcts.reform.sscs.config.NotificationEventTypeLists.EVENTS_WITH_SUBSCRIPTION_TYPE_EMAIL_TEMPLATES;
+import static uk.gov.hmcts.reform.sscs.config.PersonalisationConfiguration.PersonalisationKey.DISABLED_ACCESS;
+import static uk.gov.hmcts.reform.sscs.config.PersonalisationConfiguration.PersonalisationKey.HEARING_LOOP;
+import static uk.gov.hmcts.reform.sscs.config.PersonalisationConfiguration.PersonalisationKey.LANGUAGE_INTERPRETER;
+import static uk.gov.hmcts.reform.sscs.config.PersonalisationConfiguration.PersonalisationKey.OTHER_ARRANGEMENTS;
+import static uk.gov.hmcts.reform.sscs.config.PersonalisationConfiguration.PersonalisationKey.SIGN_INTERPRETER;
+import static uk.gov.hmcts.reform.sscs.config.PersonalisationMappingConstants.*;
```
Comment (amoslv1) [inline src/main/java/uk/gov/hmcts/reform/sscs/personalisation/Personalisation.java:46]: .*

File: src/e2e/java/uk/gov/hmcts/reform/sscs/functional/sya/notifications/DocmosisWithGovNotifyLetterFunctionalTest.java [hunk 26]
```diff
@@ -1,40 +1,91 @@
 package uk.gov.hmcts.reform.sscs.functional.sya.notifications;
 
-import static org.junit.Assert.assertEquals;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatNoException;
+import static uk.gov.hmcts.reform.sscs.config.NotificationEventTypeLists.DOCMOSIS_LETTERS;
 import static uk.gov.hmcts.reform.sscs.domain.notify.NotificationEventType.*;
```
Comment (amoslv1) [inline src/e2e/java/uk/gov/hmcts/reform/sscs/functional/sya/notifications/DocmosisWithGovNotifyLetterFunctionalTest.java:6]: .*

**Review by amoslv1 — CHANGES_REQUESTED** [review]
just check the FOR_FOR names

**Review by amoslv1 — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/service/NotificationService.java (hunk 9 above)
Comment (TeaWithLucas) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/NotificationService.java:316]: I thought isNoOrNull covered blanks, but that was in my YesNo Reactor (https://tools.hmcts.net/jira/browse/SSCS-10240) that got put on the backburner due to S&L 
That field should never be blank, only ever "Yes", "No" or null as per the ccd defs model, but to be safe ill change it to !isYes()

File: src/test/java/uk/gov/hmcts/reform/sscs/service/OutOfHoursCalculatorTest.java (hunk 10 above)
Comment (TeaWithLucas) [reply src/test/java/uk/gov/hmcts/reform/sscs/service/OutOfHoursCalculatorTest.java:58]: I changed it to a assertj assertion

File: src/test/java/uk/gov/hmcts/reform/sscs/service/OutOfHoursCalculatorTest.java (hunk 11 above)
Comment (TeaWithLucas) [reply src/test/java/uk/gov/hmcts/reform/sscs/service/OutOfHoursCalculatorTest.java:70]: assertj

File: src/e2e/java/uk/gov/hmcts/reform/sscs/functional/sya/notifications/DocmosisWithGovNotifyLetterFunctionalTest.java (hunk 6 above)
Comment (TeaWithLucas) [reply src/e2e/java/uk/gov/hmcts/reform/sscs/functional/sya/notifications/DocmosisWithGovNotifyLetterFunctionalTest.java:70]: Removed

File: src/e2e/java/uk/gov/hmcts/reform/sscs/functional/sya/notifications/DocmosisWithGovNotifyLetterFunctionalTest.java (hunk 5 above)
Comment (TeaWithLucas) [reply src/e2e/java/uk/gov/hmcts/reform/sscs/functional/sya/notifications/DocmosisWithGovNotifyLetterFunctionalTest.java:52]: Removed

**Review by amoslv1 — APPROVED** [review]

**Review by filippo-possenti-version1 — APPROVED** [review]
Approved. Please take note that 127 changed files is a huge PR. Force-push should probably be avoided on such large PRs.

**Pull request merged**